.github/workflows/go-test-multiplatform.yml

```diff
@@ -76,57 +54,90 @@ jobs:
           go-version: ${{ needs.get-go-version.outputs.go-version }}
       - run: |
           echo "Testing with Go ${{ needs.get-go-version.outputs.go-version }}"
-          filename=$(ls | grep -E '^[0-9]{12}_packer_test_report\.json')
-
-          go test -race -count 1 ./... -timeout=3m -v
+          go test -race -count 1 ./... -timeout=3m
 
   linux-go-tests:
     needs:
       - get-go-version
-      - get-test-dependencies
     runs-on: ubuntu-latest
+    env:
+      EXIT_STATUS: 0
     name: Linux Go tests
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v4
         with:
           go-version: ${{ needs.get-go-version.outputs.go-version }}
-      - run: |
-          echo "Testing with Go ${{ needs.get-go-version.outputs.go-version }}"
-          filename=$(ls | grep -E '^[0-9]{12}_packer_test_report\.json')
 
-          go test -race -count 1 ./... -timeout=3m -v -json | tee "$filename"
+      - name: Install go-junit-report
+        run: go install github.com/jstemmer/go-junit-report/v2@latest
 
-      - name: Convert JSON Report to XML
+      - name: Run integration tests
         run: |
-          filename=$(ls | grep -E '^[0-9]{12}_packer_test_report\.json')
+          timestamp=$(date +'%Y%m%d%H%M')
+          report_filename="${timestamp}_packer_test_report.xml"  
+          echo "Testing with Go ${{ needs.get-go-version.outputs.go-version }}"
+          if ! go test -race -count 1 ./... -timeout=3m | go-junit-report -set-exit-code > "$report_filename"; then
+          echo "EXIT_STATUS=1" >> $GITHUB_ENV
+          fi
+          cat "$report_filename"
           
-          if [ -f "$filename" ]; then
-            go_junit_report_dir=$(go env GOPATH)/bin
-            export PATH="$PATH:$go_junit_report_dir"
-            xml_filename=$(echo "$filename" | sed 's/\.json$/.xml/')
-            go-junit-report < "$filename" > "$xml_filename"
-            echo "Conversion from JSON to XML completed successfully."
+      - name: Upload test report
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-reports
+          path: ${{ github.workspace }}/${timestamp}_packer_test_report.xml
+
+      - name: Test Execution Status Handler
+        run: |
+          if [[ "$EXIT_STATUS" != 0 ]]; then
+            echo "Test execution contains failure(s), check Run Integration tests step above"
+            exit $EXIT_STATUS 
           else
-            echo "JSON test report file not found."
-            exit 1
+            echo "Tests passed!"
           fi
-        env:
-          GO111MODULE: on
 
+  process-upload-report:
```

Comment:
Could we ensure this only runs on pushed to `main` or `dev` since the OBJ secrets aren't accessible from PRs?